lib/logstash/inputs/lumberjack.rb

```diff
@@ -99,15 +101,14 @@ def run(output_queue)
   def invoke(connection, codec, &block)
     @threadpool.post do
       begin
-        # If any errors occur in from the events the connection should be closed in the 
+        # If any errors occur in from the events the connection should be closed in the
         # library ensure block and the exception will be handled here
-        connection.run do |fields|
-          block.call(codec, fields.delete("line"), fields)
-        end
+        client = ConnectionDecorator.new(EventDecoration.new, code, connection)
```

Comment:
I did try in the past, have problems testing this reliably. Due to this block running inside the threadpool the tests will hang for me.